frappe/public/js/frappe/views/reports/query_report.js

```diff
@@ -53,6 +53,11 @@ frappe.views.QueryReport = class QueryReport extends frappe.views.BaseList {
 			label: __('Refresh'),
 			action: () => this.refresh()
 		};
+		
+		this.report_action = {
+			label: __('Run In Background'),
```

Comment:
this is not very user friendly. The decision to run automatically should be made by the system based on certain "best guesses"

Edit 1: Or in query report, we can specify if this report is to be run in the background only. This seems to be a better decision.

Edit 2: In the query report, we already tag the reference doctype, so we can run a `count` query on the reference table and if its bigger than a number, push it to background.

Reply:
Hi Rushab,

For Edit 1: A checkbox has already been added in report doctype to mark reports to be run in background mode by default. This 'Run In Background' was added to have a submit form button on report view, so that form submission happens after all filters are set by user. But yes, this should not show up for all reports and should only be shown on reports with the background flag set in report definition. 

Also is there a better way to prevent report from triggering before user has entered all the filters ?

For Edit-2 : This functionality has to be available to all types for reports (Query, Script, Standard), so count query might not be feasible. Also the whole intention of building this is to run reports that take several minutes, hours and sometimes days also :). So, a count query is also very expensive on these tables most of the time and also the report might not be just dependent on the table at all.

Reply:
Cool, then lets remove this option. If its a prepared report, then just show download button for the last report (with the relevant filters) with the timestamp or show status of the current one under preparation.

Count is not required if its a prepared report.

For filters, you can add a validation on the server side to make sure that the report is only triggered with filters.

(maybe we should have filters as a child table in the report, so we can control them more effectively)